### DISCUSSION TITLE
Multiple uv_write triggered only once uv_read_cb

### DISCUSSION THREAD

**Discussion by l1046j** [libuv/libuv#3298] [Q&A]
I called uv_write multiple times and sent multiple messages, but uv_read_cb was called only once and I read all the messages sent before. I want to trigger uv_read_cb every time I call uv_write. What should I do?
* **1.42**: 
* **Platform**:

**Comment by l1046j** [comment]
Windows 10

**Comment by bnoordhuis** [comment]
Your problem description is a bit light on details but if you're talking about TCP or pipes, the answer is "you don't" - they're byte streams (stress on "streams"), not packet oriented protocols. Your operating system's network stack can (and will) coalesce packets upon receiving them.

**Reply by l1046j** [reply]
TCP. I hope to recive uv_read_cb callback(on server) for each uv_write(on client). if I use uv_sleep between each uv_write,it`s ok, but I don`t hope so

**Reply by deniszykov** [reply]
make you own frames with [bodyLength][body] is more reliable and performant way to solve your problem.
